### PR TITLE
Optimise file operations to avoid use of the file server

### PR DESCRIPTION
**Pull request by kjnilsson**
Avoid any file based functions that use the file server in favour of "raw" versions.

### CODE REVIEW

File: src/ra.app.src [hunk 1]
```diff
@@ -1,6 +1,6 @@
 {application,ra,
              [{description,"Raft library"},
-              {vsn,"2.0.1"},
+              {vsn,"2.0.2"},
```
Comment (michaelklishin) [inline src/ra.app.src:3]: Should this be `2.0.3` since `2.0.2` has already shipped?

Comment (kjnilsson) [reply src/ra.app.src:3]: No for some reason this commit went into the wrong branch. Main has 2.0.1

File: src/ra_lib.erl [hunk 2]
```diff
@@ -337,6 +347,72 @@ lists_take(_N, [], Acc) ->
...
+    case prim_file:read_file_info(Dir) of
+        {ok, #file_info{type=directory}} ->
+            true;
+        _ ->
+            false
+    end.
+
+is_file(File) ->
+    case prim_file:read_file_info(File) of
+        {ok, #file_info{type = directory}} ->
```
Comment (michaelklishin) [inline src/ra_lib.erl:361]: Just a note: this behavior is consistent with that of `filelib`: https://www.erlang.org/doc/man/filelib.html#is_file-1

**Comment by michaelklishin** [comment]
One interesting side effect of this: the faster the (parallel) QQ recovery process gets, the higher will be peak file handle use on the node.

**Review by michaelklishin — APPROVED** [review]

**Pull request merged**